core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
The only misuse I see is in `cb_onboard` which as far as I know is a deprecated, unused module. All the other instances of map/fold use concrete keys, built from the version in cb_context. Not to throw Pierre D. under the bus here, but all instances of using a wildcard in a routing key were done by him (and we failed to catch it in review).

kazoo_bindings is an attempt to have local AMQP-like routing - would you publish an AMQP message with a wildcard in the routing key? kazoo_bindings does have an optimization for crossbar bindings wherein kazoo_bindings uses the first two segments of the binding key to create a pared-down result set when comparing against a routing key (otherwise we have to use the whole ETS table of binding keys to see which match).

Modules that bind should be able to clean up after themselves by using the same binding key(s) they used when calling bind/3. flush/1 takes a binding key and flushes all responders bound to that key. flush_mod/1 takes a module and checks all bindings' responders for that module, removing it from the responders (and removing the binding if responders is empty).

If there are binding keys registered that need cleaning up, that's a failing of the application that put them there. So I would ask, why does `tasks.#` need cleaning up at startup? Why is it there in the first place, who put it there, and why didn't they clean it up?